crates/builder/src/main.rs

```diff
@@ -1,103 +1,60 @@
 #![allow(dead_code)]
 
-mod env;
-use env::LocalOrAws;
-
+mod config;
 mod service;
+mod signer;
 mod tasks;
 
 use alloy_network::EthereumSigner;
-use alloy_network::TxSigner;
-use alloy_primitives::Address;
 use alloy_provider::ProviderBuilder;
-use std::borrow::Cow;
 use tokio::select;
 use zenith_types::Zenith;
 
-use crate::env::{load_quincey, load_rpc, load_zenith_address};
+use crate::config::load_builder_config;
 use crate::service::serve_builder_with_span;
-
-/// Configuration for a builder running a specific rollup on a specific host
-/// chain.
-pub struct ChainConfig {
-    /// The chain ID of the host chain
-    pub host_chain_id: u64,
-    /// The chain ID of the host chain
-    pub ru_chain_id: u64,
-    /// Buffer time in seconds for the block to confirm
-    pub confirmation_buffer: u64,
-    /// address of the Zenith contract
-    pub zenith: Address,
-    /// URL for Quincey server to sign blocks. This prop is disregarded if a
-    /// local_sequencer_signer is configured via the "SEQUENCER_KEY" env var.
-    pub quincey_url: Cow<'static, str>,
-    /// URL for RPC node
-    pub rpc_url: Cow<'static, str>,
-
-    /// Wallet for signing blocks locally.
-    pub local_sequencer_signer: Option<LocalOrAws>,
-
-    /// Whether to use calldata or blob for transactions
-    pub use_calldata: bool,
-}
-
-impl ChainConfig {
-    /// Creates a new ChainConfig for the Holesky testnet.
-    const fn holesky(
-        local_sequencer_signer: Option<LocalOrAws>,
-        rpc_url: Cow<'static, str>,
-        quincey_url: Cow<'static, str>,
-        zenith: Address,
-    ) -> Self {
-        ChainConfig {
-            host_chain_id: 17000,
-            ru_chain_id: 17001,
-            confirmation_buffer: 60 * 20,
-            zenith,
-            quincey_url,
-            rpc_url,
-            local_sequencer_signer,
-            use_calldata: true,
-        }
-    }
-}
+use crate::signer::LocalOrAws;
 
 #[tokio::main]
 async fn main() -> eyre::Result<()> {
     tracing_subscriber::fmt::try_init().unwrap();
-
     let span = tracing::info_span!("zenith-builder");
 
-    // finish app config by loading key from env
-    let config = ChainConfig::holesky(None, load_rpc()?, load_quincey()?, load_zenith_address()?);
-
-    // Load builder key from env
-    let wallet = LocalOrAws::load("BUILDER_KEY_ID", Some(config.host_chain_id)).await?;
-    let builder_rewards = wallet.address();
+    // load config from env
+    let config = load_builder_config()?;
 
+    // build provider from config
+    let builder_signer = LocalOrAws::load(&config.builder_key, Some(config.host_chain_id)).await?;
```

Comment:
I would like this line to be 
`let builder_signer = config.connect_builder_signer().await?;`

which means a function 

```
impl BuilderConfig {
    /// Connect the configured builder signer 
    pub async fn connect_builder_signer(&self) -> Result<LocalOrAws, ConfigError>;
}

Reply:
https://github.com/init4tech/zenith-rs/pull/12/commits/8b0e0d49f02fe66f0cd3fb9dab3876731fc519cb